src/mnode/ticket-processor.cpp

```diff
@@ -675,6 +691,94 @@ std::string CPastelTicketProcessor::ListFilterTradeTickets(const short filter) c
         });
 }
 
+/*static*/ bool CPastelTicketProcessor::WalkBackTradingChain(
```

Comment:
This is very cool function to traverse trading chain!

Can simplify this and remove lots of returns like this:

bool bRet = false;
do
{
    .....


      if (pastelTicket-&gt;ID() == TicketID::Trade)   {
          auto tradeTicket = dynamic_cast<CArtTradeTicket*>(pastelTicket.get());
          if (!tradeTicket)
          { 
                errRet = strprintf("The Trade ticket [txid=%s] referred by this ticket [txid=%s] is invalid",                               
                           pastelTicket->GetTxId(), sTxId);
                break;
           }
           if (!WalkBackTradingChain(shortPath? tradeTicket->artTnxId: tradeTicket->buyTnxId, chain, shortPath, errRet))            
                break; 
     }
     .......
     // in case everything succeeds here we can change bRet to true
    chain.emplace_back(std::move(pastelTicket));
    bRet = true;
} while (false);
return bRet;

Reply:
good advice, yes - i like do{}while for long logic